src/rust/bitbox02/src/queue.rs

```diff
@@ -1,3 +1,23 @@
-pub fn pull_hww() -> *const u8 {
-    unsafe { bitbox02_sys::queue_pull(bitbox02_sys::queue_hww_queue()) }
+// Copyright 2025 Shift Crypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+pub fn pull_hww() -> Option<[u8; 64]> {
+    let hww_data = unsafe { bitbox02_sys::queue_pull(bitbox02_sys::queue_hww_queue()) };
+    if hww_data.is_null() {
+        return None;
+    }
+    let mut data: [u8; 64] = [0; 64];
+    let slice: &[u8] = unsafe { std::slice::from_raw_parts(hww_data, 64) };
+    data.copy_from_slice(slice);
+    Some(data)
```

Comment:
`Some(slice.try_into().unwrap())` would also work if you prefer. Strange there is no from_raw_parts for arrays with const generics yet.

Reply:
Hmm, now that I think about it I even prefer to use `ptr::copy_nonoverlapping` and never even instantiate the "questionable" slice.